libminifi/include/utils/file/FileManager.h

```diff
@@ -61,46 +62,25 @@ class FileManager {
     }
   }
   std::string unique_file(const std::string &location, bool keep = false) {
+    const std::string& dir = !IsNullOrEmpty(location) ? location : utils::file::FileUtils::get_temp_directory();
 
-	 
-    if (!IsNullOrEmpty(location)) {
-      std::string file_name = location + FILE_SEPARATOR + non_repeating_string_generator_.generate();
-      while (!verify_not_exist(file_name)) {
-        file_name = location + FILE_SEPARATOR + non_repeating_string_generator_.generate();
-      }
-      if (!keep)
-        unique_files_.push_back(file_name);
-      return file_name;
-    } else {
-	  std::string tmpDir = "/tmp";
-	  #ifdef WIN32
-			TCHAR lpTempPathBuffer[MAX_PATH];
-			GetTempPath(MAX_PATH, lpTempPathBuffer);
-			tmpDir = lpTempPathBuffer;
-	  #endif
-      std::string file_name = tmpDir + FILE_SEPARATOR + non_repeating_string_generator_.generate();
-      while (!verify_not_exist(file_name)) {
-        file_name = tmpDir + FILE_SEPARATOR + non_repeating_string_generator_.generate();
-      }
-      if (!keep)
-        unique_files_.push_back(file_name);
-      return file_name;
+    std::string file_name = utils::file::FileUtils::concat_path(dir, non_repeating_string_generator_.generate());
+    while (!verify_not_exist(file_name)) {
+      file_name = utils::file::FileUtils::concat_path(dir, non_repeating_string_generator_.generate());
     }
+    if (!keep)
+      unique_files_.push_back(file_name);
+    return file_name;
   }
 
   std::string unique_file(bool keep = false) {
 #ifdef BOOST_VERSION
     return boost::filesystem::unique_path().native();
 #else
-	  std::string tmpDir = "/tmp";
-	#ifdef WIN32
-		  TCHAR lpTempPathBuffer[MAX_PATH];
-		  GetTempPath(MAX_PATH, lpTempPathBuffer);
-		  tmpDir = lpTempPathBuffer;
-	#endif
-    std::string file_name = tmpDir + FILE_SEPARATOR + non_repeating_string_generator_.generate();
+	  std::string tmpDir = utils::file::FileUtils::get_temp_directory();
+    std::string file_name = utils::file::FileUtils::concat_path(tmpDir, non_repeating_string_generator_.generate());
     while (!verify_not_exist(file_name)) {
-      file_name = tmpDir + FILE_SEPARATOR + non_repeating_string_generator_.generate();
+      file_name = utils::file::FileUtils::concat_path(tmpDir, non_repeating_string_generator_.generate());
     }
     if (!keep)
       unique_files_.push_back(file_name);
```

Comment:
To me this function looks like it could nicely delegate.
```
  std::string unique_file(bool keep = false) {
#ifdef BOOST_VERSION
    return boost::filesystem::unique_path().native();
#else  // BOOST_VERSION
    return unique_file(std::string{}, keep);
#endif  // BOOST_VERSION
  }
```

Reply:
done